### DISCUSSION TITLE
Can i bind existing Rich theme to textual app?

### DISCUSSION THREAD

**Discussion by SanctusAnimus** [Textualize/textual#2618] [Q&A]
I have a few Rich tables and panels that share common theme styles, and i'm passing them to render in TextLog (as well as some Labels etc).
I haven't found a way to pass existing theme into the app, for it to highlight properly, so it's currently throwing errors at me.
Am i missing something?

**Comment by mrossinek** [comment]
I also just ran into this as I was trying to use a custom `rich.Theme` for the renderables that get passed into textual.
I found a way to do this by doing the following:
```python
from rich.theme import Theme
from textual.app import App

theme = Theme({
    "myprop": "red",
})

class MyApp(App):
    def __init__(*args, **kwargs):
        super().__init__(*args, **kwargs):
        self.console.push_theme(theme)
```

The [rich docs](https://rich.readthedocs.io/en/stable/reference/console.html#rich.console.Console.push_theme) state one should favor `use_theme` over `push_theme` but that does not work because we need the theme to basically always be in use (or at least whenever textual renders something...)
